openhands/llm/llm.py

```diff
@@ -337,6 +325,19 @@ def is_caching_prompt_active(self) -> bool:
             )
         )
 
+    def is_function_calling_active(self) -> bool:
+        # Check if model name is in supported list before checking model_info
+        model_name_supported = (
+            self.config.model in FUNCTION_CALLING_SUPPORTED_MODELS
+            or self.config.model.split('/')[-1] in FUNCTION_CALLING_SUPPORTED_MODELS
+            or any(m in self.config.model for m in FUNCTION_CALLING_SUPPORTED_MODELS)
+        )
+        return (
+            model_name_supported
+            and self.model_info is not None
+            and self.model_info.get('supports_function_calling', False)
```

Comment:
Just wondering, will `model_info` be always non-None, for Anthropic models via "openai/" proxy?